framework/src/engine/generator/endpoint.ts

```diff
@@ -171,19 +171,19 @@ export class Endpoint {
 			ctx.logger.debug(`Last generated information does not exist for address: ${req.address}`);
 		}
 
-		if (lastGeneratedInfo !== undefined && !isEqualGeneratedInfo(req, lastGeneratedInfo)) {
+		if (lastGeneratedInfo && !isEqualGeneratedInfo(req, lastGeneratedInfo)) {
 			throw new Error('Request does not match last generated information.');
 		}
-		if (lastGeneratedInfo === undefined && !isZeroValueGeneratedInfo(req)) {
-			throw new Error('Last generated information does not exist.');
-		}
-
-		if (lastGeneratedInfo === undefined) {
-			await setLastGeneratedInfo(
-				generatorStore,
-				cryptoAddress.getAddressFromLisk32Address(req.address),
-				req,
-			);
+		if (!lastGeneratedInfo) {
+			if (isZeroValueGeneratedInfo(req)) {
+				await setLastGeneratedInfo(
+					generatorStore,
+					cryptoAddress.getAddressFromLisk32Address(req.address),
+					req,
+				);
+			} else {
+				throw new Error('Last generated information does not exist.');
```

Comment:
instead of if/else, original code looks a lot more cleaner to me 🤔

Reply:
Ahhh, I see why. I didn't like the overlapping conditionals, but in this case one of them is used for throwing if the case is invalid, before taking the action.

I've reverted the change, and to avoid confusion I moved setting last generated info after setting the keypairs.